ntoskrnl/mm/ARM3/sysldr.c

```diff
@@ -2964,7 +2964,52 @@ MmLoadSystemImage(IN PUNICODE_STRING FileName,
     PrefixName = *FileName;
 
     /* Check if we have a prefix */
-    if (NamePrefix) DPRINT1("Prefixed images are not yet supported!\n");
+    if (NamePrefix)
+    {
+        /* Check if "directory + prefix" is too long for the string */
+        if(PrefixName.MaximumLength + BaseDirectory.Length < PrefixName.MaximumLength)
+        {
+            Status = STATUS_INSUFFICIENT_RESOURCES;
+            goto Quickie;
+        }
+
+        /* Set maximum length to the size of "directory + prefix" */
+        PrefixName.MaximumLength += BaseDirectory.Length;
```

Comment:
I see what you are doing here (with the previous if-test): you are checking whether the addition is going to overflow.
I wonder whether we already have the following function, that you could use there:
https://learn.microsoft.com/en-us/windows-hardware/drivers/ddi/ntintsafe/nf-ntintsafe-rtlushortadd

Reply:
I have changed the checks to use `RtlUShortAdd` now.